src/expr/src/scalar/mod.rs

```diff
@@ -2424,9 +2424,12 @@ impl EvalError {
             EvalError::InvalidBase64EndSequence => Some(
                 "Input data is missing padding, is truncated, or is otherwise corrupted.".into(),
             ),
-            EvalError::LikeEscapeTooLong => {
-                Some("Escape string must be empty or one character.".into())
-            }
+            EvalError::LikeEscapeTooLong => Some(
+                "Escape string must be empty or one character.".into()
+            ),
+            EvalError::MzTimestampOutOfRange => Some(
+                "Integer, numeric, and text casts to mz_timestamp must be in the form of milliseconds since the Unix epoch.".into()
```

Comment:
I think we need to be more specific by saying something like "form of whole milliseconds" and/or adding a sentence:

> Values with fractional parts cannot be converted to mz_timestamp

One (maybe bad) thought I had was to add a `from: String` field to `EvalError::MzTimestampOutOfRange` that would contain the `.to_string()` of the failing conversion so users could more clearly see the failing datum?

Reply:
> Values with fractional parts cannot be converted to mz_timestamp

Good suggestion. I will add.

> One (maybe bad) thought I had was to add a from: String field to EvalError::MzTimestampOutOfRange that would contain the .to_string() of the failing conversion so users could more clearly see the failing datum?

Yeah I had that thought too - it's a best practice in case the cast is nested in a bigger statement. Perhaps we should be doing that for all casting errors, e.g. `Float32OutOfRange`...? I can file a separate follow up ticket to add the field to all of those. There are ~20 others that should be updated.

Reply:
https://github.com/MaterializeInc/materialize/issues/16219